viewer/src/main/java/org/kaazing/example/agrona/MetricsViewer.java

```diff
@@ -1,58 +1,49 @@
 
-import uk.co.real_logic.agrona.IoUtil;
-import uk.co.real_logic.agrona.concurrent.CountersManager;
 import uk.co.real_logic.agrona.concurrent.SigInt;
-import uk.co.real_logic.agrona.concurrent.UnsafeBuffer;
 
 public class MetricsViewer {
 
-    private static int COUNTER_LABELS_BUFFER_LENGTH = 32 * 1024 * 1024;
-    private static int COUNTER_VALUES_BUFFER_LENGTH = 1024 * 1024;
-    private static final String LINUX_OS = "Linux";
-    private static final String OS_NAME_SYSTEM_PROPERTY = "os.name";
-    private static final String LINUX_DEV_SHM_DIRECTORY = "/dev/shm";
-	
+    private static final int UPDATE_INTERVAL = 1000;
 
     public static void main(String[] args) throws InterruptedException {
-        String prefix = "";
-        if (LINUX_OS.equalsIgnoreCase(System.getProperty(OS_NAME_SYSTEM_PROPERTY))) {
-              prefix = LINUX_DEV_SHM_DIRECTORY;
-        }
-        File tmpDir = new File(prefix + IoUtil.tmpDirName() + "/kaazing", "monitor");
-        MappedByteBuffer mapNewFile = IoUtil.mapExistingFile(tmpDir, "monitor");
 
-        UnsafeBuffer labelsBuffer = new UnsafeBuffer(mapNewFile, 64, COUNTER_LABELS_BUFFER_LENGTH);
-        UnsafeBuffer valuesBuffer =
-                new UnsafeBuffer(mapNewFile, 64 + COUNTER_LABELS_BUFFER_LENGTH, COUNTER_VALUES_BUFFER_LENGTH);
+        MetricsCollector metricsCollector = MetricsCollectorFactory.getInstance();
 
-        CountersManager countersManager = new CountersManager(labelsBuffer, valuesBuffer);
+        if (metricsCollector == null) {
+            System.out.println("There was a problem initializing the metrics reader. Exiting application.");
+            System.exit(1);
+        }
 
-        final AtomicBoolean running = new AtomicBoolean(true);
-        SigInt.register(() -> running.set(false));
+        // Waits until the metrics file is created by the producer (e.g. this app is started and then waits for a
+        // gateway to start and create the file)
+        while (!metricsCollector.initialize()) {
+            Thread.sleep(UPDATE_INTERVAL);
+        }
 
-        while (running.get()) {
+        ScheduledExecutorService taskExecutor = Executors.newScheduledThreadPool(1);
+        SigInt.register(() -> taskExecutor.shutdown());
+
+        taskExecutor.scheduleAtFixedRate(() -> {
             System.out.println("Agrona counters:");
             System.out.format("%1$tH:%1$tM:%1$tS - Gateway monitor\n", new Date());
             System.out.println("=========================");
             System.out.println("Counter id: Counter value - Counter name");
 
-            countersManager.forEach((id, label) -> {
-                final int offset = CountersManager.counterOffset(id);
-                final long value = valuesBuffer.getLongVolatile(offset);
-
-                System.out.format("%3d: %,20d - %s\n", id, value, label);
-            });
+            metricsCollector.getMetrics();
 
             System.out.println("");
```

Comment:
Is there a plan to eventually move to a logging system, e.g. java.util.logging or log4j?